kqueen_ui/utils/forms.py

```diff
@@ -48,7 +48,9 @@ def append_fields(cls, ctx, switchtag=None):
                 }
             }
         '''
-        fields = OrderedDict(sorted(ctx.items(), key=lambda k: k[0]))
+        # Sort engine parameters by order if set and by name in the second turn
+        fields = OrderedDict(sorted(ctx.items(), key=lambda k: (k[1].get('order', 0), k[0])))
```

Comment:
On a second though, you could use `label` for secondary sorting, because it may be translable in the future and you want to sort by the actual first letter which is shown on the screen, which may change with translation. So maybe use:
 `fields = OrderedDict(sorted(ctx.items(), key=lambda k: (k[1].get('order', 0), k[1].get('label'))))`. I just hope that `label` is always present

Reply:
PS: it may be possible to default to key in case of mising label like this: `get('label', k[0])` ... it will get a bit too complex, but should be the best way to handle this sorting

Reply:
Remember, the main idea in alphabet sorting is to keep order from one form load to another, and order is present where it is needed. So I suggest not to make sorting more complicated and leave it at it is - sort by order or by name in parameter schema key

Reply:
OK